### PR TITLE
Added Pagination Tests

### PR DESCRIPTION
**Pull request by MuminAhmadKhan**
Added Test for pagination functionality

### CODE REVIEW

File: __mocks__/mockarticles.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        image: "https://assets.guim.co.uk/images/eada8aa27c12fe2d5afa3a89d3fbae0d/fallback-logo.png",
+        region: "Europe"
+        },
+        'PN7QJPEP4K': {
+            title: "Payback time: eco upgrades based on the new energy price cap",
+            url: "https://12ft.io/https://www.thetimes.co.ukhttps://www.thetimes.co.uk/article/how-to-save-energy-at-home-reduce-energy-bills-8bv763fn6",
+            source: "The Times",
+            image: "https://www.thetimes.co.uk/d/assets/society-69df0d1a8d.png",
+            region: "Europe"
+            },
```
Comment (MizouziE) [inline __mocks__/mockarticles.js:57]: The formatting is a little weird in this file **BUT** it is still valid JSON so I shouldn't complain. :sweat_smile:

File: __tests__/unit-tests/paginate.test.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    it('first page should have next and no previous if total is greater than  limit ', () => {
+        const value = paginate(mockReq, mockRes, mockarticles);
+        //console.log(value)
+        expect(value.next).toBeDefined();
+        expect(value.previous).toBeUndefined();
+        
+    });
+    it('last page should have no next  ', () => {
+        const value = paginate({...mockReq,query:{limit:5,page:Math.ceil(Object.keys(mockarticles).length/5)}}, mockRes, mockarticles);
+       // console.log(value)
```
Comment (MizouziE) [inline __tests__/unit-tests/paginate.test.js:81]: Also this one, thanks!

**Review by MizouziE — CHANGES_REQUESTED** [review]
The actual tests are great and I see in the CI that they all pass so :+1: :+1: 

Just a tiny little bit of tidy up and it's good to go.

Thanks for writing these! :pray:

File: __tests__/unit-tests/paginate.test.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        if ( ariclesLength< mockReq.query.limit)
+        {
+        expect(Object.keys(value.results).length).toBe(ariclesLength);
+        }
+        else  expect(Object.keys(value.results).length).toBe(parseInt(mockReq.query.limit));
+        
+    });
+    it('first page should have no next and no previous if total is less than or equal to limit ', () => {
+        const value = paginate(mockReq, mockRes,Object.fromEntries(Object.entries(mockarticles).slice(0, mockReq.query.limit)));
+        //console.log(value)
```
[the remark this answers is not in the document]
Comment (MuminAhmadKhan) [reply __tests__/unit-tests/paginate.test.js:67]: This test or just the console.log?

[the remark this answers is not in the document]
Comment (MizouziE) [reply __tests__/unit-tests/paginate.test.js:67]: Just the console.log()

The tests are good.

[the remark this answers is not in the document]
Comment (MuminAhmadKhan) [reply __tests__/unit-tests/paginate.test.js]: Thanks for the feedback and sorry for such silly mistakes.
Please check now.

**Review by MizouziE — APPROVED** [review]
Thanks!

**Pull request merged**